### ISSUE TITLE
[Training] Strange training model created with crossEntropyLoss

### ISSUE THREAD

**Issue by pauldog** [microsoft/onnxruntime#17875]
### Describe the issue

Trying out the training model creation on the MNIST-12 from [here](https://github.com/onnx/models/tree/main/vision/classification/mnist/model), when I set the loss to artifacts.LossType.CrossEntropyLoss,
it does something strange.

Even though the output label is of shape (1,10), it creates an input node called "labels" with shape of size (1)

Surely the labels shape should also be of size (1,10) ?

When using artifacts.LossType.MSELoss is creates an input node called "target" of shape (1,10) which is correct.

Is this a bug?

![snapshot5](https://github.com/microsoft/onnxruntime/assets/33497043/448a7ab7-dd0a-4a04-841b-a4b2ab2e7aa1)

**Edit:**
I just realised that perhaps the label is not of the form (0,0,0,1,0,0,0,0,0,0) but just the index 4, in this case. Is this correct? 

OK, I think I get it. you can delete this post.

### To reproduce

AS above

### ONNX Runtime Installation

Released Package

### ONNX Runtime Version or Commit ID

1.15.1

### PyTorch Version

1.13.1

### Execution Provider

CUDA

### Execution Provider Library Version

1.13.1

**Comment by baijumeswani** [comment]
> I just realised that perhaps the label is not of the form (0,0,0,1,0,0,0,0,0,0) but just the index 4, in this case. Is this correct?

This is correct. Thanks for checking. Closing this now.
